NMS-14615: Quick Start: Import inventory (#5244)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/provisioning/foreign-source.adoc

[[foreign-source-definition]]
= Foreign Source Definition

The default foreign source definition serves as a template that defines the services to detect on (DNS, FTP, ICMP, and so on), the scan interval for discovery, and the policies to use when provisioning.

> Policies determine entity persistence and/or set attributes on the discovered entities that control {page-component-title} management behavior.

Reviewer: ```suggestion
Policies determine entity persistence and set attributes on the discovered entities that control {page-component-title} management behavior.
```

Provisiond applies the existing default foreign source definition unless you choose to modify it.

This section describes how to edit the default foreign source definition.

== Edit detectors and set scan interval

. In the web UI, click the *Gear* symbol in the top-right of the screen.
. Choose menu:Provisioning[Manage Provisioning Requisitions > Edit Default FS].
+
image::provisioning/FS_def.png[]
+
NOTE: The screen displays the list of service detectors and a tab to view and define policies.
Provisiond scans the services in the order in which the detectors appear in the list.

** Click the *Edit* symbol to edit a service detector's parameters.
To add more parameters (retries, timeout, port, and so on), click *Add Parameter* and define its name and value.
** Click the *Move* symbol to move a service detector.
** Click the *Delete* symbol to delete a service detector.
** To undo a change, click *Reset to Default* at the top-right of the screen.
. Click *Save*.
. (Optional) Update the scan interval using one of the following:
+
* w: weeks
* d: days
* h: hours
* m: minutes
* s: seconds
* ms: milliseconds
+

> For example, to prompt a scan every 6 days and 53 minutes, use `6d53m`.

Reviewer: ```suggestion
For example, to prompt a scan every 6 days and 53 minutes, enter `6d53m`.
```

Type `0` to disable automatic scanning.

. Click *Save* to save the foreign source definition.

== Create policies

. In the web UI, click the *Gear* symbol in the top-right of the screen.
. Choose menu:Provisioning[Manage Provisioning Requisitions > Edit Default FS] and click the Policies tab.
. Click *Add Policy*, type a name, select the class from the *Class* list, and enter the required information.
** Type a space in any of the lists to see their options.
. (Optional) Click *Add Parameter* to add more parameters to the class (retries, timeout, port, and so on).
. Click *Save*.
. Repeat for any additional policies you want to add.
. Click *Save* to save the foreign source definition.

> NOTE: To return to the default foreign source definition, click *Reset to Default*.

Reviewer: ```suggestion
NOTE: To return to the default foreign source definition, click *Reset to Default*.
This will reset any changes that you have made to the definition.
```

[[foreign-source-definition]]
= Foreign Source Definition

The default foreign source definition serves as a template that defines the services to detect on (DNS, FTP, ICMP, and so on), the scan interval for discovery, and the policies to use when provisioning.

Policies determine entity persistence and/or set attributes on the discovered entities that control {page-component-title} management behavior.
Provisiond applies the existing default foreign source definition unless you choose to modify it.

This section describes how to edit the default foreign source definition.

== Edit detectors and set scan interval

> . In the web UI, click the gear icon in the top right.

Reviewer: ```suggestion
. In the web UI, click the *Gear* symbol in the top-right of the screen.
```

> . In the *Provisioning* area, choose *Manage Provisioning Requisitions*.
> . Click *Edit Default FS*.
> 
> +

Reviewer: ```suggestion
. Choose menu:Provisioning[Manage Provisioning Requisitions > Edit Default FS].
+
```

> image::provisioning/FS_def.png[]
> 
> +

Reviewer: ```suggestion
image::provisioning/FS_def.png[]
+
```

NOTE: The screen displays the list of service detectors and a tab to view and define policies.
Provisiond scans the services in the order in which the detectors appear in the list.

> . Click the appropriate icon to edit, delete, or move a service detector.
> .. You can also add parameters to a detector, including retries, timeout, port, and so on, by clicking the Edit icon and choosing *Add Parameter*.

Reviewer: ```suggestion
** Click the *Edit* symbol to edit a service detector's parameters.
To add more parameters (retries, timeout, port, and so on), click *Add Parameter* and define its name and value.
** Click the *Move* symbol to move a service detector.
** Click the *Delete* symbol to delete a service detector.
** If you make a change that you want to undo, click *Reset to Default* at the top-right of the screen.
```

Author: Reading this, we probably need to add a warning that clicking Reset to Default removes all of your changes. (Will test and verify.) If this is the case, someone could lose all the configuration they have done for 17 detectors ... I wonder if there is a way they can back it up.

. Click *Save*.

> . If desired, update the scan interval using one of the following:
> 
> +
> * w: weeks
> * d: days
> * h: hours
> * m: minutes
> * s: seconds
> * ms: milliseconds
> 
> +
> For example, to rescan every six days and 53 minutes, use `6d 53m`.
> Specify `0` to disable automatic scanning.

Reviewer: ```suggestion
. (Optional) Update the scan interval using one of the following:
+
* w: weeks
* d: days
* h: hours
* m: minutes
* s: seconds
* ms: milliseconds
+
For example, to prompt a scan every 6 days and 53 minutes, use `6d53m`.
Type `0` to disable automatic scanning.
```

> . Click *Save*.

Reviewer: ```suggestion
. Click *Save* to save your changes.
```

== Create policies

> . Click the *Policies* tab in the *Default Foreign Source Definition* screen.

Reviewer: ```suggestion
. In the web UI, click the *Gear* symbol in the top-right of the screen.
. Choose menu:Provisioning[Manage Provisioning Requisitions > Edit Default FS] and click the Policies tab.
```

> . Specify a name for the policy, select the class from the drop down, and fill out any information associated with that class.

Reviewer: ```suggestion
. Click *Add Policy*, type a name, select the class from the Class list, and enter the required information.
```

> .. Use the space bar to see the options for the fields.

Reviewer: ```suggestion
** Type a space in any of the lists to see their options.
```

> . (optional) Click *Add Parameter* to add additional parameters to the class, or *Save*.

Reviewer: ```suggestion
. (Optional) Click *Add Parameter* to add more parameters to the class (retries, timeout, port, and so on).
```

. Click *Save*.
. Repeat for any additional policies you want to add.

> . Click *Save* at the top right to save the FS definition.

Reviewer: ```suggestion
. Click *Save* to save the foreign source definition.
```
